From d58a32776e08e193c24f7bb192daf72b66fbe550 Mon Sep 17 00:00:00 2001
From: Peter Seiderer <ps.report@gmx.net>
Date: Wed, 18 Jul 2018 21:35:18 +0200
Subject: [PATCH] meson.build: enable static build

Use meson library instead of shared_library to enable static build.

Upstream suggested: https://lists.freedesktop.org/archives/dri-devel/2018-July/183886.html
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 amdgpu/meson.build    | 2 +-
 etnaviv/meson.build   | 2 +-
 exynos/meson.build    | 2 +-
 freedreno/meson.build | 2 +-
 intel/meson.build     | 2 +-
 libkms/meson.build    | 2 +-
 meson.build           | 2 +-
 nouveau/meson.build   | 2 +-
 omap/meson.build      | 2 +-
 radeon/meson.build    | 2 +-
 tegra/meson.build     | 2 +-
 11 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/amdgpu/meson.build b/amdgpu/meson.build
index f39d7bf..1b4b0be 100644
--- a/amdgpu/meson.build
+++ b/amdgpu/meson.build
@@ -21,7 +21,7 @@

 datadir_amdgpu = join_paths(get_option('prefix'), get_option('datadir'), 'libdrm')

-libdrm_amdgpu = shared_library(
+libdrm_amdgpu = library(
   'drm_amdgpu',
   [
     files(
diff --git a/etnaviv/meson.build b/etnaviv/meson.build
index ca2aa54..a0d994e 100644
--- a/etnaviv/meson.build
+++ b/etnaviv/meson.build
@@ -19,7 +19,7 @@
 # SOFTWARE.


-libdrm_etnaviv = shared_library(
+libdrm_etnaviv = library(
   'drm_etnaviv',
   [
     files(
diff --git a/exynos/meson.build b/exynos/meson.build
index 30d3640..fd14f3a 100644
--- a/exynos/meson.build
+++ b/exynos/meson.build
@@ -18,7 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.

-libdrm_exynos = shared_library(
+libdrm_exynos = library(
   'drm_exynos',
   [files('exynos_drm.c', 'exynos_fimg2d.c'), config_file],
   c_args : warn_c_args,
diff --git a/freedreno/meson.build b/freedreno/meson.build
index 015b7fb..6c8a6a7 100644
--- a/freedreno/meson.build
+++ b/freedreno/meson.build
@@ -39,7 +39,7 @@ if with_freedreno_kgsl
   )
 endif

-libdrm_freedreno = shared_library(
+libdrm_freedreno = library(
   'drm_freedreno',
   [files_freedreno, config_file],
   c_args : warn_c_args,
diff --git a/intel/meson.build b/intel/meson.build
index 53c7fce..14cabd3 100644
--- a/intel/meson.build
+++ b/intel/meson.build
@@ -18,7 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.

-libdrm_intel = shared_library(
+libdrm_intel = library(
   'drm_intel',
   [
     files(
diff --git a/libkms/meson.build b/libkms/meson.build
index 86d1a4e..9d21e4c 100644
--- a/libkms/meson.build
+++ b/libkms/meson.build
@@ -41,7 +41,7 @@ if with_exynos
   libkms_include += include_directories('../exynos')
 endif

-libkms = shared_library(
+libkms = library(
   'kms',
   [files_libkms, config_file],
   c_args : warn_c_args,
diff --git a/meson.build b/meson.build
index 3c8afb6..7aa5f8c 100644
--- a/meson.build
+++ b/meson.build
@@ -279,7 +279,7 @@ add_project_arguments('-include', 'config.h', language : 'c')
 inc_root = include_directories('.')
 inc_drm = include_directories('include/drm')

-libdrm = shared_library(
+libdrm = library(
   'drm',
   [files(
      'xf86drm.c', 'xf86drmHash.c', 'xf86drmRandom.c', 'xf86drmSL.c',
diff --git a/nouveau/meson.build b/nouveau/meson.build
index 51c9a71..acba048 100644
--- a/nouveau/meson.build
+++ b/nouveau/meson.build
@@ -19,7 +19,7 @@
 # SOFTWARE.


-libdrm_nouveau = shared_library(
+libdrm_nouveau = library(
   'drm_nouveau',
   [files( 'nouveau.c', 'pushbuf.c', 'bufctx.c', 'abi16.c'), config_file],
   c_args : warn_c_args,
diff --git a/omap/meson.build b/omap/meson.build
index e57b8f5..6cffb99 100644
--- a/omap/meson.build
+++ b/omap/meson.build
@@ -18,7 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.

-libdrm_omap = shared_library(
+libdrm_omap = library(
   'drm_omap',
   [files('omap_drm.c'), config_file],
   include_directories : [inc_root, inc_drm],
diff --git a/radeon/meson.build b/radeon/meson.build
index b08c744..2f45ff7 100644
--- a/radeon/meson.build
+++ b/radeon/meson.build
@@ -19,7 +19,7 @@
 # SOFTWARE.


-libdrm_radeon = shared_library(
+libdrm_radeon = library(
   'drm_radeon',
   [
     files(
diff --git a/tegra/meson.build b/tegra/meson.build
index 1f5c74b..8d0cfa9 100644
--- a/tegra/meson.build
+++ b/tegra/meson.build
@@ -18,7 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.

-libdrm_tegra = shared_library(
+libdrm_tegra = library(
   'drm_tegra',
   [files('tegra.c'), config_file],
   include_directories : [inc_root, inc_drm],
--
2.18.0

